### PR TITLE
add scramble pin toggle

### PR DESCRIPTION
**Pull request by deregs**
# Description

Adds toggle in Security settings to choose whether pin numbers should be scrambled.

Because all current users' pins are scrambled, made it so that default state is `scrambledPin = true`

Also changed the Security Settings view to use a class instead of a function in order to remain consistent with the rest of the views, and also to make it easier to inject `SettingsStore`. Without injecting `SettingsStore` I was running into bugs.

This pull request is categorized as a:

- [x] New feature
- [ ] Bug fix
- [ ] Code refactor
- [ ] Configuration change
- [ ] Locales update
- [ ] Other

## Checklist
- [x] I’ve run `npm run tsc` and made sure my code compiles correctly
- [x] I’ve run `npm run lint` and made sure my code didn’t contain any problematic patterns
- [x] I’ve run `npm run prettier` and made sure my code is formatted correctly
- [x] I’ve run `npm run test` and made sure all of the tests pass

## Testing

If you added new functionality or fixed a bug, did you add new unit tests?

- [ ] No, I’m a fool
- [ ] Yes
- [x] N/A

I have tested this PR on the following platforms (please specify OS version and phone model/VM):

- [ ] Android
- [x] iOS

I have tested this PR with the following types of nodes (please specify node version and API version where appropriate):

- [ ] LND
- [ ] c-lightning-REST
- [ ] spark
- [ ] Eclair
- [ ] LndHub

### Locales
- [ ] I’ve added new locale text that requires translations
- [ ] I’m aware that new translations should be made on the Zeus [Transfix page](https://www.transifex.com/ZeusLN/zeus/) and not directly to this repo

### Third Party Dependencies and Packages

- [ ] Contributors will need to run `npm install` after this PR is merged in
- [ ] 3rd party dependencies have been modified:
    * verify that `package.json` and `package-lock.json` have been properly updated
    * verify that dependencies are installed for both iOS and Android platforms

### Other:

- [ ] Changes were made that require an update to the README
- [ ] Changes were made that require an update to onboarding

### CODE REVIEW

File: views/Settings/AddEditNode.tsx [hunk 1]
```diff
@@ -340,24 +341,11 @@ export default class AddEditNode extends React.Component<
             enableTor
         };
 
-        setSettings(
```
Comment (kaloudis) [inline views/Settings/AddEditNode.tsx:343]: why is `setSettings` ripped out here?

Comment (deregs) [reply views/Settings/AddEditNode.tsx:343]: There's no purpose to this call as far as I can see, we're just getting the settings and setting it to the exact same settings. Which makes sense, when copying the node config nothing in the settings changes. The copied node is only added to settings once save node config is pressed on the copied configuration.

File: views/Settings/Security.tsx [hunk 2]
```diff
@@ -133,35 +154,97 @@ function Security(props: SecurityProps) {
...
+                    }}
+                >
+                    <ListItem.Content>
+                        <ListItem.Title
+                            style={{
+                                color: themeColor('secondaryText'),
+                                fontFamily: 'Lato-Regular'
+                            }}
+                        >
+                            {'Scramble PIN numbers'}
```
Comment (kaloudis) [inline views/Settings/Security.tsx:210]: This should be added to `locales/en.json`

**Comment by Bosch-0** [comment]
I'd recommend this be off by default and have it opt-in for a better UX

File: views/Settings/Security.tsx (hunk 2 above)
Comment (deregs) [reply views/Settings/Security.tsx]: Done

**Comment by deregs** [comment]
> 

I agree, but discussed with @kaloudis and we thought for now it should default to scrambled. Since all current users have scramble PIN on, we want to avoid defaulting to non-scrambled and unscrambling current users' pins. Ideally, we could design out a create pin flow that includes the scramble pin toggle so the user can just decide scrambled/non-scrambled off the bat. It also might make more sense to users if they could toggle scrambled on/off during pin creation and see the numbers on the pin pad get scrambled/unscrambled and decide which they prefer.

**Review by kaloudis — APPROVED** [review]

**Pull request merged**